File: original java project files/Appointment/AppointmentService.java
/*
 * Name: Justin Fifarek
 * Course: SNHU CS-320
 * Date: 12/01/2023
 * 
 * Module 5-1 Milestone - Appointment Service class
 */

package Appointment;

import java.util.ArrayList;
import java.util.Date;

public class AppointmentService {
	// Array list to store appointments
	public ArrayList<Appointment> appointmentList = new ArrayList<Appointment>();
	
	// Display all appointments in the list
	public void displayApptList() {
		for (int i = 0; i < appointmentList.size(); i++) {
			System.out.println("Appointment ID: " + appointmentList.get(i).getAppointmentID());
			System.out.println("Appointment Date: " + appointmentList.get(i).getAppointmentDate());
			System.out.println("Appointment Description: " + appointmentList.get(i).getAppointmentDesc());
		}
	}
	
	// Adds a new appointment using the constructor 
	// and then adds to the appointment list
	public void addAppointment(Date appointmentDate, String appointmentDesc) {
		// Create the new appointment
		Appointment appointment = new Appointment(appointmentDate, appointmentDesc);
		appointmentList.add(appointment);
	}
	
	// Returns an appointment given appointmentID, if it exists. 
	// Otherwise, returns a default appointment
	public Appointment getAppointment(String appointmentID) {
		Appointment appointment = new Appointment(null, null);
		for (int i = 0; i < appointmentList.size(); i++) {
			if (appointmentList.get(i).getAppointmentID().contentEquals(appointmentID)) {
				appointment = appointmentList.get(i);
			}
		}
		return appointment;
	}
	
	// Loops through appointmentList for given appointmendID 
	// and deletes it if found.
	public void deleteAppointment(String appointmentID) {
		for (int i = 0; i < appointmentList.size(); i++) {
			if (appointmentList.get(i).getAppointmentID().equals(appointmentID)) {
				appointmentList.remove(i);
				break;
			}
			if (i == appointmentList.size() - 1) {
				System.out.println("Appointment ID: " + appointmentID + " not found.");
			}
		}
	}

	// Updates appointmentDate of appointmentID given a newDate
	public void updateAppointmentDate(Date newDate, String appointmentID) {
		for (int i = 0; i < appointmentList.size(); i++) {
			if (appointmentList.get(i).getAppointmentID().equals(appointmentID)) {
				appointmentList.get(i).setAppointmentDate(newDate);
				break;
			}
			if (i == appointmentList.size() - 1) {
				System.out.println("Appointment ID: " + appointmentID + " not found.");
			}
		}
	}

	// Updates appointmentDesc of given appointmendID
	public void updateAppointmentDesc(String newDesc, String appointmentID) {
		for (int i = 0; i < appointmentList.size(); i++) {
			if (appointmentList.get(i).getAppointmentID().equals(appointmentID)) {
				appointmentList.get(i).setAppointmentDesc(newDesc);
				break;
			}
			if (i == appointmentList.size() - 1) {
				System.out.println("Appointment ID: " + appointmentID + " not found.");
			}
		}
	}
}

File: docs/module5/apt-management-app/src/AppointmentService.cpp
/*
    Name: Justin Fifarek
    Project: Appointment management app
    Module: AppointmentService class file
*/

#include "../includes/AppointmentService.h"

// Display all appointments in the list
void AppointmentService::displayAppointments() {
    for (auto appointment : appointmentList) {
        std::cout << "Appointment ID: " << appointment.getAppointmentID() << std::endl;
        std::cout << " Appointment Date: " << appointment.getAppointmentDate() << std::endl;
        std::cout << " Appointment Description: " << appointment.getAppointmentDesc() << std::endl;
    }
}

// Display given appointmentID from appointmentList
bool AppointmentService::displayOneAppointment(unsigned int appointmentID) {
    for (auto appointment : appointmentList) {
        if (appointment.getAppointmentID() == appointmentID) {
            std::cout << "Appointment ID: " << appointment.getAppointmentID() << std::endl;
            std::cout << " Appointment Date: " << appointment.getAppointmentDate() << std::endl;
            std::cout << " Appointment Description: " << appointment.getAppointmentDesc() << std::endl;
            for (size_t x = 0; x < appointment.getAppointmentDesc().length(); ++x) {
                std::cout << "-";
            }
            std::cout << std::endl;
            return true;
        }
    }
    return false;
}

// Adds new appointment using the appointment constructor
void AppointmentService::addAppointment(std::string appointmentDate, std::string appointmentDesc) {
    // Create the new appointment
    Appointment appointment(appointmentDate, appointmentDesc);
    appointmentList.push_back(appointment);
}

// Returns an appointment given appointmentID, if it exists. 
// Otherwise, returns a default appointment
Appointment AppointmentService::getAppointment(unsigned int appointmentID) {
    for (auto appointment : appointmentList) {
        if (appointment.getAppointmentID() == appointmentID) {
            return appointment;
        }
    }
    return Appointment("", "");
}

// Loops through appointmentList for given appointmentID 
// and deletes it if found.
void AppointmentService::deleteAppointment(unsigned int appointmentID) {
    for (auto it = appointmentList.begin(); it != appointmentList.end(); ++it) {
        if (it->getAppointmentID() == appointmentID) {
            appointmentList.erase(it);
            return;
        }
    }
    std::cout << "Appointment ID: " << appointmentID << " not found." << std::endl;
}

// Updates appointmentDate of appointmentID given a newDate
void AppointmentService::updateAppointmentDate(std::string newDate, unsigned int appointmentID) {
    for (auto appointment : appointmentList) {
        if (appointment.getAppointmentID() == appointmentID) {
            appointment.setAppointmentDate(newDate);
            return;
        }
    }
    std::cout << "Appointment ID: " << appointmentID << " not found." << std::endl;
}

// Updates appointmentDesc of given appointmentID
void AppointmentService::updateAppointmentDesc(std::string newDesc, unsigned int appointmentID) {
    for (auto appointment : appointmentList) {
        if (appointment.getAppointmentID() == appointmentID) {
            appointment.setAppointmentDesc(newDesc);
            return;
        }
    }
    std::cout << "Appointment ID: " << appointmentID << " not found." << std::endl;
};